src/main/java/fi/okm/jod/yksilo/controller/ehdotus/MahdollisuudetController.java

```diff
@@ -120,28 +135,62 @@ public List<EhdotusDto> createEhdotus(@RequestBody @Valid LuoEhdotusDto ehdotus)
                 ehdotus.kiinnostusPainotus,
                 kiinnostukset.stream().map(o -> o.nimi().get(Kieli.FI)).toList()));
 
-    var result =
-        inferenceService.infer(endpoint, request, Response.class).stream()
-            .collect(Collectors.toMap(Suggestion::id, r -> r, (exising, newValue) -> exising));
+    return inferenceService.infer(endpoint, request, Response.class).stream()
+        .collect(Collectors.toMap(Suggestion::id, r -> r, (exising, newValue) -> exising));
+  }
+
+  private List<EhdotusDto> populateEmptyEhdotusDtos(
+      LinkedHashSet<UUID> ids, Set<UUID> tyomahdollisuusIds) {
 
+    // initialize lexicalIndex counters
+    final var tmLexicalIndex = new AtomicInteger();
+    final var kmLexicalIndex = new AtomicInteger();
```

Comment:
Using AtomicInteger seems unnecessary, the stream processing is sequential so there are no thread safety issues.